util/fhir.ts

```diff
@@ -34,6 +35,36 @@ export function createPatientResourceString(birthDate: string): string {
   return JSON.stringify(pt, null, 2);
 }
 
+/**
+ * Identifies the primary code path of a resource and constructs a string which displays
+ * resource summary information depending on what is a available
+ * @param resource {Object} a fhir Resource object
+ * @returns {String} displaying the code and display text, code, or id of the resource or nothing
+ */
+export function getFhirResourceSummary(resource: fhir4.Resource) {
```

Comment:
This function should have some unit tests that test it directly instead of through the component. The tests should be much easier to write to test all the paths through this function.